File: Core.API/ViewModels/Token.cs
using Core.Models;
using System;
using System.Collections.Generic;

namespace Core.ViewModels
{
    [Serializable]
    public partial class Token
    {

        public int UserId { get; set; }
        public int? CostCenterId { get; set; }
        public string UserName { get; set; }
        public string Email { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullName { get; set; }
        public string Address { get; set; }
        public string Avatar { get; set; }
        public string AccessToken { get; set; }
        public string RefreshToken { get; set; }
        public DateTime AccessTokenExp { get; set; }
        public DateTime RefreshTokenExp { get; set; }
        public string HashPassword { get; set; }
        public string Recovery { get; set; }
        public string SysName { get; set; } = "TMS";
        public string TenantCode { get; set; }
        public Vendor Vendor { get; set; }
        public List<int> RoleIds { get; set; }
        public List<string> RoleNames { get; set; }
        public List<int> AllRoleIds { get; set; }
        public List<int> CenterIds { get; set; }
        public string Ssn { get; set; }
        public string PhoneNumber { get; set; }
        public int? TeamId { get; set; }
        public string PartnerId { get; set; }
        public int? RegionId { get; set; }
        public object Additional { get; set; }
    }

    [Serializable]
    public class GPS
    {
        public double latitude { get; set; }
        public double longitude { get; set; }
    }

    public class RefreshVM
    {
        public string AccessToken { get; set; }
        public string RefreshToken { get; set; }
    }
}

File: CoreAPI/wwwRoot/js/core/models/token.js
import { Client } from "../clients/client.js";

/**
 * Represents an authentication token with user details and permissions.
 */
export class Token {
    /** @type {string|null} User's unique identifier */
    UserId = null;

    /** @type {string|null} Identifier for the user's cost center */
    CostCenterId = null;

    /** @type {string|null} User's username */
    UserName = null;

    /** @type {string|null} User's email address */
    Email = null;

    /** @type {string|null} User's first name */
    FirstName = null;

    /** @type {string|null} User's last name */
    LastName = null;

    /** @type {string|null} User's full name */
    FullName = null;

    /** @type {string|null} User's address */
    Address = null;

    /** @type {string|null} URL to the user's avatar image */
    Avatar = null;

    /** @type {string|null} Access token for user authentication */
    AccessToken = null;

    /** @type {string|null} Refresh token for renewing the access token */
    RefreshToken = null;

    /**
     * Expiration time of the access token.
     * @type {Date|null}
     */
    AccessTokenExp = new Date();

    /**
     * Expiration time of the refresh token.
     * @type {Date|null}
     */
    RefreshTokenExp = new Date();

    /** @type {string|null} Hashed password for the user */
    HashPassword = null;

    /** @type {string|null} Recovery token for password reset */
    Recovery = null;

    /**
     * Vendor information associated with the user.
     * @type {Vendor|null}
     */
    Vendor = null;

    /** @type {Array<string>|null} List of role identifiers for the user */
    RoleIds = [];

    /** @type {Array<string>|null} List of role names for the user */
    RoleNames = [];

    /** @type {Array<string>|null} List of center identifiers for the user */
    CenterIds = [];

    /** @type {string|null} User's social security number */
    Ssn = null;

    /** @type {string|null} User's phone number */
    PhoneNumber = null;

    /** @type {string|null} Identifier for the user's team */
    TeamId = null;

    /** @type {string|null} Identifier for the user's partner entity */
    PartnerId = null;

    /** @type {string|null} Identifier for the user's regional entity */
    RegionId = null;

    /** @type {object|null} Additional arbitrary data associated with the user */
    Additional = null;

    /**
     * Date and time the user signed in.
     * @type {Date|null}
     */
    SigninDate = new Date();

    /** @type {string|null} Tenant code for the user's tenant */
    TenantCode = null;

    /** @type {string|null} Environment context for the user session */
    Env = null;

    /** @type {string|null} Connection key used for database connections */
    ConnKey = null;

    constructor() {
        // Default values can be initialized here if different from null or empty.
        this.TenantCode = Client.Tenant; // Assuming Client.Tenant is accessible
        this.Env = Client.Env;          // Assuming Client.Env is accessible
        this.ConnKey = Client.MetaConn; // Assuming Client.MetaConn is accessible
    }
}
